### PR TITLE
powsimp correctly handles sign extraction from signed base

### LINKED ISSUE #17524: Wrong result from simplify()
Simplify two identical matrix gives different results:

```
import sympy
import sys
from sympy import *
init_printing()
print("Sympy version : %s" % sympy.__version__)
l = symbols(r"a", real=True)
U = Matrix([
    [l, -1/l, -1/l],
    [0, sqrt(-l**2-1)/l, -I*sqrt(l**2+1)/l],
    [1, 1, 1],
])
Uinv = U.inv()
display(U)
display(simplify(Uinv*U))
display(simplify(U*Uinv))
```

Results with sympy 1.2 or 1.4:

`Uinv*U` gives   `[[1,0,0],[0,1,0],[0,0,1]]` 
`U*Uinv` gives   `[[1,-ia^2/sqrt(1+a^2),0],[0,1,0],[0,0,1]]` , which is wrong

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #17524 

#### Brief description of what is fixed or changed

During base combination in `powsimp` the exponent corresponding to the base being removed from `c_powers` is now being used. A base raised to an integer power can always add to `(-base)**x` as can a positive base raised to any power. In these cases base `b` and exponent `e` are used.

When a base is negative then the negated base (which is positive) and *its* exponent is used.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/simplify/powsimp.py [hunk 1]
```diff
@@ -203,8 +203,14 @@ def recurse(arg, **kwargs):
         _n = S.NegativeOne
         for i, (b, e) in enumerate(be):
```
Comment (jksuom) [inline sympy/simplify/powsimp.py]: `i` does not seem to be used.

File: sympy/simplify/powsimp.py [hunk 2]
```diff
@@ -203,8 +203,14 @@ def recurse(arg, **kwargs):
...
             if ((-b).is_Symbol or b.is_Add) and -b in c_powers:
-                if (b.is_positive in (0, 1) or e.is_integer):
+                if (b.is_positive is not None or e.is_integer):
                     c_powers[-b] += c_powers.pop(b)
```
Comment (jksuom) [inline sympy/simplify/powsimp.py]: Shouldn't we pop `-b` if `b` is positive?

**Comment by oscarbenjamin** [comment]
> I need someone to take a look at the Order docstring/test changes.

I took a look and this looks good to me but I don't know Order so well...

File: sympy/series/tests/test_order.py [hunk 3]
```diff
@@ -88,7 +88,9 @@ def test_simple_8():
     assert O(x**2*sqrt(x)) == O(x**(S(5)/2))
     assert O(x**3*sqrt(-(-x)**3)) == O(x**(S(9)/2))
     assert O(x**(S(3)/2)*sqrt((-x)**3)) == O(x**3)
-    assert O(x*(-2*x)**(I/2)) == O(x*(-x)**(I/2))
+    # not sure why the first doesn't become the 3rd without
+    # iteration in the Order.__new__ routine.
+    assert O(x*(-2*x)**(I/2)) == O(x*(-x)**(I/2)) == O((-x)**(1 + I/2))
```
Comment (jksuom) [inline sympy/series/tests/test_order.py]: It seems that the order algorithm should be improved. By definition, `(-2*x)**(I/2)` is `exp(log(-2*x)*I/2)`. Assuming that the the principal branch of the logarithm should be used, the imaginary part of `log(-2*x)` is between `-pi` and `pi`. Hence the real part of `log(-2*x)*I/2` is between `-pi/2` and `pi/2`, and therefore the absolute value of `(-2*x)**(I/2)` is in the bounded interval `[exp(-pi/2), exp(pi/2)]`. This means the the order of `x*(-2*x)**(I/2)` is `O(x)`.

Comment (jksuom) [reply sympy/series/tests/test_order.py]: I think it is ok to commit this. Fixing Order is another matter and independent of powsimp. (It looks like imaginary exponents should have no effect on order.)

**Pull request merged**